packages/ubuntu_desktop_installer/test/connect_to_internet_page_test.dart

```diff
@@ -0,0 +1,193 @@
+    expect(model.connectMode, ConnectMode.wifi);
+
+    final noConnectTile = find.byWidgetPredicate((widget) =>
+        widget is RadioButton<ConnectMode> && widget.value == ConnectMode.none);
+    expect(noConnectTile, findsOneWidget);
+    await tester.tap(noConnectTile);
+    expect(model.connectMode, ConnectMode.none);
+  });
+
+  testWidgets('pre-selects ethernet', (tester) async {
+    final model = ConnectToInternetModel(MockNetworkService());
+    await buildTestApp(tester, model: model, ethernetActive: true);
+    await tester.pumpAndSettle();
+
+    final ethernetTile = find.byType(EthernetRadioButton);
+    expect(ethernetTile, findsOneWidget);
+    expect(tester.widget<EthernetRadioButton>(ethernetTile).value,
+        ConnectMode.ethernet);
+    expect(model.connectMode, ConnectMode.ethernet);
+  });
+
+  testWidgets('pre-selects wifi', (tester) async {
+    final model = ConnectToInternetModel(MockNetworkService());
+    await buildTestApp(tester, model: model, wifiActive: true);
+    await tester.pumpAndSettle();
+
+    final wifiTile = find.byType(WifiRadioButton);
+    expect(wifiTile, findsOneWidget);
+    expect(tester.widget<WifiRadioButton>(wifiTile).value, ConnectMode.wifi);
+    expect(model.connectMode, ConnectMode.wifi);
+  });
+
+  testWidgets('pre-selects no connect', (tester) async {
+    final model = ConnectToInternetModel(MockNetworkService());
+    await buildTestApp(tester,
+        model: model, ethernetActive: false, wifiActive: false);
+    await tester.pumpAndSettle();
+
+    final noConnectTile = find.byWidgetPredicate((widget) =>
+        widget is RadioButton<ConnectMode> && widget.value == ConnectMode.none);
+    expect(noConnectTile, findsOneWidget);
+    expect(model.connectMode, ConnectMode.none);
+  });
+
+  testWidgets('creates all models', (tester) async {
+    final service = MockNetworkService();
+    when(service.propertiesChanged)
+        .thenAnswer((_) => const Stream<List<String>>.empty());
+    when(service.wiredDevices).thenReturn([]);
+    when(service.wirelessDevices).thenReturn([]);
+    when(service.wirelessEnabled).thenReturn(true);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        localizationsDelegates: localizationsDelegates,
+        home: Scaffold(
+          body: MultiProvider(
+            providers: [
+              Provider<NetworkService>.value(value: service),
+              Provider<UdevService>(create: (_) => MockUdevService()),
+            ],
+            child: const Wizard(
+              routes: {'/': WizardRoute(builder: ConnectToInternetPage.create)},
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final page = find.byType(ConnectToInternetPage);
+    expect(page, findsOneWidget);
+
+    final context = tester.element(page);
+    expect(
+        Provider.of<ConnectToInternetModel>(context, listen: false), isNotNull);
+    expect(
+        Provider.of<ConnectToInternetModel>(context, listen: false), isNotNull);
```

Comment:
This line is duplicated, I suppose it should test the existence of a `NoConnectModel` instance instead?